src/pages/YourPeep/YourOffChainPeep.tsx

```diff
@@ -0,0 +1,145 @@
+	const classes = useStyles()
+	const { metadata, setMetadata } = useContext(MetadataContext)
+	const { profile, setProfile } = useContext(ProfileContext)
+	const [yourPeepImage, setYourPeepImage] = useState('')
+	const [milkyGreeting, setMilkyGreeting] = useState('Aloha')
+	const [jonoGreeting, setJonoGreeting] = useState('Ciao')
+	const [grannyGreeting, setGrannyGreeting] = useState('Hello')
+	const [officerGreeting, setOfficerGreeting] = useState('Bonjour!')
+
+	if (
+		!metadata ||
+		!setMetadata ||
+		!profile ||
+		!setProfile ||
+		!useLocation().state
+	) {
+		return <></>
+	}
+
+	const getRandomGreeting = () => {
+		const i = Math.floor(Math.random() * greetings.length)
+		return greetings[i]
+	}
+	const { uri } = useLocation().state as propState
+
+	const changePhrase = (setGreeting: (text: string) => void) => {
+		setGreeting(getRandomGreeting())
+	}
+
+	useEffect(() => {
+		const getYourPeep = async () => {
+			if (uri) {
+				const peepImage = await getPeepFromURI(uri)
+				setYourPeepImage(peepImage)
+
+				// reset metadata in case they want to mint another
+				setMetadata([...DEFAULT_PEEP])
+				setProfile({ address: profile.address })
+			}
+		}
+		getYourPeep()
+	}, [])
+
+	const getPeepFromURI = async (peepURI: string) => {
+		const svg = await doFetch(
+			`${host}/peep/${peepURI}.svg?offchain=true`,
+			'GET',
+			undefined,
+			'image/svg+xml',
+		)
+		return URL.createObjectURL(svg)
+	}
+
+	return (
+		<div className={classes.page}>
+			<h2 className={classes.title}>
+				You look amazing
+				{getTrait(metadata, 'Name')}!!
+			</h2>
+			<img className={classes.peepImage} src={yourPeepImage} />
+			<div>
+				<p>Your Peep's ID is important, keep it safe.</p>
+				<Copy text={uri} />
+			</div>
+			<div className={classes.buttonGroup}>
+				<Link className={classes.button} to={HomeRoute.path}>
+					Make Another?
+				</Link>
+			</div>
+
+			<div className={classes.milky}>
+				{milkyGreeting.length > 0 && (
+					<div
+						className={`${classes.bubbleLeft} ${classes.fadeInAndOut1}`}
+						onAnimationIteration={() => {
+							changePhrase(setMilkyGreeting)
+						}}
+					>
+						{milkyGreeting}
+					</div>
+				)}
+				<img className={classes.guestPeep} src={'/assets/Milky.svg'} />
+			</div>
+			<div className={classes.jono}>
+				{jonoGreeting.length > 0 && (
+					<div
+						className={`${classes.bubbleLeft} ${classes.fadeInAndOut2}`}
+						onAnimationIteration={() => {
+							changePhrase(setJonoGreeting)
+						}}
+					>
+						{jonoGreeting}
+					</div>
+				)}
+				<img className={classes.guestPeep} src={'/assets/Jono.svg'} />
+			</div>
+			<div className={classes.granny}>
+				{grannyGreeting.length > 0 && (
+					<div
+						className={`${classes.bubbleRight} ${classes.fadeInAndOut3}`}
+						onAnimationIteration={() => {
+							changePhrase(setGrannyGreeting)
+						}}
+					>
+						{grannyGreeting}
+					</div>
+				)}
+				<img className={classes.guestPeep} src={'/assets/Granny Peep.svg'} />
```

Comment:
Can we make these a component? `BouncingPeep` and pass in the fade class, greeting, greeting setter, svg path